lib/openapi_parser/schema_validator/string_validator.rb

```diff
@@ -8,7 +8,18 @@ def initialize(validator, coerce_value, datetime_coerce_class)
     end
 
     def coerce_and_validate(value, schema, **_keyword_args)
-      return OpenAPIParser::ValidateError.build_error_result(value, schema) unless value.kind_of?(String)
+      unless value.kind_of?(String)
+        # Skip validation if the format is `binary`, even if the value is not an actual string.
+        # ref: https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#data-types
+        if schema.format == 'binary'
```

Comment:
This seems preferable to an error, but I worry a bit about this in effect now silently failing instead. Maybe as a middle ground we could add something like: `warn('[WARNING] skipping binary validation (not yet implemented)`. I'm open to suggestions on the specific wording. This way it isn't a blocker to people using it, but it should also make it really clear to them that it isn't validating.

Otherwise I worry that people might falsely believe it is being validated, even though it isn't, which could create it's own problems. What do you think?

Reply:
Thank you for your suggestion.

> Maybe as a middle ground we could add something like: warn('[WARNING] skipping binary validation (not yet implemented). 

This sounds good to me, and I don't have a strong opinion on it.

However, I assume that "binary" means just that, and the OpenAPI document describes it as "any sequence of octets." Thus, I'm guessing that this implies we might not have to validate the data format or apply any data encapsulation/decapsulation.
An API client does not necessarily use an API solely for file uploading (e.g., a user might need to send a truly arbitrary binary stream). If so, omitting data validation is natural rather than treating it specially, as in `Rack::Multipart::UploadFile` or another special type.

Reply:
Good point, thanks for talking through this. Let's go ahead with this.